src/sentry_plugins/twilio/plugin.py

```diff
@@ -136,37 +141,34 @@ def notify_users(self, group, event, **kwargs):
         )
         body = body[:MAX_SMS_LENGTH]
 
-        account_sid = self.get_option("account_sid", project)
-        auth_token = self.get_option("auth_token", project)
-        sms_from = clean_phone(self.get_option("sms_from", project))
-        endpoint = twilio_sms_endpoint.format(account_sid)
+        client = self.get_client(group.project)
 
-        sms_to = self.get_option("sms_to", project)
-        if not sms_to:
-            return
-        sms_to = split_sms_to(sms_to)
-
-        headers = {"Authorization": basic_auth(account_sid, auth_token)}
+        payload = {"From": client.sms_from, "Body": body}
 
         errors = []
 
-        for phone in sms_to:
+        for phone in client.sms_to:
             if not phone:
                 continue
             try:
                 phone = clean_phone(phone)
-                http.safe_urlopen(
-                    endpoint,
-                    method="POST",
-                    headers=headers,
-                    data={"From": sms_from, "To": phone, "Body": body},
-                ).raise_for_status()
+                payload["To"] = phone
+                client.request(payload)
             except Exception as e:
                 errors.append(e)
 
         if errors:
-            if len(errors) == 1:
-                raise errors[0]
+            self.raise_error(errors[0])
 
             # TODO: multi-exception
             raise Exception(errors)
+
+    def get_client(self, project):
+        account_sid = self.get_option("account_sid", project)
+        auth_token = self.get_option("auth_token", project)
+        sms_from = clean_phone(self.get_option("sms_from", project))
+        sms_to = self.get_option("sms_to", project)
+        if not sms_to:
+            return
```

Comment:
@ceorourke Rather than returning `None` here if the plugin is not configured, we should call `is_configured` at the top of `notify_users` and quit at that point instead of here. See what's done in OpsGenie.